zng/recordval.go

```diff
@@ -335,6 +320,16 @@ func (r *Record) AccessTimeByColumn(colno int) (nano.Ts, error) {
 	return DecodeTime(zv)
 }
 
+// Ts returns the value of the receiver's "ts" field.  If the field is absent,
+// is null, or has a type other than TypeOfTime, Ts returns nano.MinTs.
+func (r *Record) Ts() nano.Ts {
+	if !r.tsValid {
+		r.ts, _ = r.AccessTime("ts")
```

Comment:
I love getting errors when we do not need them but... how do we know there will be nothing wrong with the underlying buffer when we try to access time here?

Reply:
@mattnibs: In general, we don't. In practice, if a record survives (*Record).TypeCheck, we do, and all existing zio reader implementations call that method (either directly or via NewRecordCheck).

Are you thinking we shouldn't discard the error here?

Reply:
maybe at least a panic?

Reply:
Like this?
```go
	if !r.tsValid {
		var err error
		r.ts, err = r.AccessTime("ts")
		if err != ErrNoSuchField && err != ErrTypeMismatch {
			panic(err)
		}
		r.tsValid = true
	}
```